In [23]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [24]:
df_newyork_flights = pd.read_csv('/Users/rezanaiman/Desktop/Fall 2024/Stat captson/ICR project/final_merged_2024.csv')

/var/folders/wg/8_pdnpf51ys7nljv71s3c5jw0000gn/T/ipykernel_10992/2536031708.py:1: DtypeWarning: Columns (36,37,39,43) have mixed types. Specify dtype option on import or set low_memory=False.
  df_newyork_flights = pd.read_csv('/Users/rezanaiman/Desktop/Fall 2024/Stat captson/ICR project/final_merged_2024.csv')


In [25]:
df_newyork_flights.head()

YEAR  QUARTER  MONTH  DAY_OF_MONTH  DAY_OF_WEEK               FL_DATE  \
0  2024        1      1             1            1  1/1/2024 12:00:00 AM   
1  2024        1      1             1            1  1/1/2024 12:00:00 AM   
2  2024        1      1             1            1  1/1/2024 12:00:00 AM   
3  2024        1      1             1            1  1/1/2024 12:00:00 AM   
4  2024        1      1             1            1  1/1/2024 12:00:00 AM   

  OP_UNIQUE_CARRIER  OP_CARRIER_AIRLINE_ID TAIL_NUM  OP_CARRIER_FL_NUM  ...  \
0                9E                  20363   N133EV               5284  ...   
1                9E                  20363   N133EV               5341  ...   
2                9E                  20363   N133EV               5390  ...   
3                9E                  20363   N133EV               5390  ...   
4                9E                  20363   N133EV               5390  ...   

      skyl3  skyl4 wxcodes peak_wind_gust  peak_wind_drct  peak_wind_time  \
0  13000.00      M       M              M               M               M   
1       NaN    NaN     NaN            NaN             NaN             NaN   
2       NaN    NaN     NaN            NaN             NaN             NaN   
3         M      M       M              M               M               M   
4         M      M     -RA              M               M               M   

    feel                                              metar  snowdepth  \
0  43.55  KLGA 011751Z 30003KT 10SM FEW030 BKN070 BKN130...          M   
1    NaN                                                NaN        NaN   
2    NaN                                                NaN        NaN   
3  36.97  KLGA 011051Z 29007KT 10SM BKN035 OVC060 06/M01...          M   
4  38.27  KLGA 011113Z 29005KT 10SM -RA OVC029 06/M01 A3...          M   

         rounded_valid  
0  2024-01-01 18:00:00  
1                  NaN  
2                  NaN  
3  2024-01-01 11:00:00  
4  2024-01-01 11:00:00  

[5 rows x 61 columns]

In [26]:
df_newyork_flights.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63773 entries, 0 to 63772
Data columns (total 61 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   YEAR                   63773 non-null  int64  
 1   QUARTER                63773 non-null  int64  
 2   MONTH                  63773 non-null  int64  
 3   DAY_OF_MONTH           63773 non-null  int64  
 4   DAY_OF_WEEK            63773 non-null  int64  
 5   FL_DATE                63773 non-null  object 
 6   OP_UNIQUE_CARRIER      63773 non-null  object 
 7   OP_CARRIER_AIRLINE_ID  63773 non-null  int64  
 8   TAIL_NUM               63576 non-null  object 
 9   OP_CARRIER_FL_NUM      63773 non-null  int64  
 10  ORIGIN_AIRPORT_ID      63773 non-null  int64  
 11  ORIGIN_CITY_MARKET_ID  63773 non-null  int64  
 12  ORIGIN                 63773 non-null  object 
 13  ORIGIN_STATE_ABR       63773 non-null  object 
 14  DEST_AIRPORT_ID        63773 non-null  int64  
 15  DE

In [27]:
df_newyork_flights.iloc[39648].info() #looking at the 39648 observation

<class 'pandas.core.series.Series'>
Index: 61 entries, YEAR to rounded_valid
Series name: 39648
Non-Null Count  Dtype 
--------------  ----- 
33 non-null     object
dtypes: object(1)
memory usage: 3.0+ KB


In [28]:
df_newyork_flights.iloc[39647].info()#looking at the 39647 observation

<class 'pandas.core.series.Series'>
Index: 61 entries, YEAR to rounded_valid
Series name: 39647
Non-Null Count  Dtype 
--------------  ----- 
33 non-null     object
dtypes: object(1)
memory usage: 3.0+ KB


In [29]:
df_newyork_flights.iloc[39645]#looking at the 39645 observation

YEAR              2024
QUARTER              1
MONTH                1
DAY_OF_MONTH        21
DAY_OF_WEEK          7
                  ... 
peak_wind_time     NaN
feel               NaN
metar              NaN
snowdepth          NaN
rounded_valid      NaN
Name: 39645, Length: 61, dtype: object

There are a lot of missing or na values, so we will have to remove those observations that contains na or missing values.

In [30]:
df_newyork_flights = df_newyork_flights.dropna() #dropping na values

In [31]:
df_newyork_flights.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 8862 entries, 90 to 63757
Data columns (total 61 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   YEAR                   8862 non-null   int64  
 1   QUARTER                8862 non-null   int64  
 2   MONTH                  8862 non-null   int64  
 3   DAY_OF_MONTH           8862 non-null   int64  
 4   DAY_OF_WEEK            8862 non-null   int64  
 5   FL_DATE                8862 non-null   object 
 6   OP_UNIQUE_CARRIER      8862 non-null   object 
 7   OP_CARRIER_AIRLINE_ID  8862 non-null   int64  
 8   TAIL_NUM               8862 non-null   object 
 9   OP_CARRIER_FL_NUM      8862 non-null   int64  
 10  ORIGIN_AIRPORT_ID      8862 non-null   int64  
 11  ORIGIN_CITY_MARKET_ID  8862 non-null   int64  
 12  ORIGIN                 8862 non-null   object 
 13  ORIGIN_STATE_ABR       8862 non-null   object 
 14  DEST_AIRPORT_ID        8862 non-null   int64  
 15  DE

In [32]:
df_newyork_flights.shape

(8862, 61)

After dropping the na values, now we are left 8862 observations.